eclipse/src/saros/Saros.java

```diff
@@ -200,6 +211,59 @@ public static boolean useHtmlGui() {
     return Platform.getBundle("saros.ui") != null && Boolean.getBoolean("saros.swtbrowser");
   }
 
+  private void shutdownPlugin(final boolean readAndDispatch) {
+
+    if (shutdownInitiated) return;
+
+    shutdownInitiated = true;
+
+    final AtomicBoolean isShutdownComplete = new AtomicBoolean(false);
+
+    final Display display = Display.getCurrent();
+
+    final Thread shutdownThread =
+        ThreadUtils.runSafeAsync(
+            "dpp-shutdown", //$NON-NLS-1$
+            log,
+            () -> {
+              try {
+                lifecycle.stop();
+              } finally {
+                isShutdownComplete.set(true);
+
+                if (display != null) {
+                  try {
+                    display.wake();
```

Comment:
Is this `display.wake()` required in the `if` case?
The `readAndDispatch` loop is never entered and we don't have to wake the display? Am I wrong?